tests/cypress/tests/unit/formatting_test.ts

```diff
@@ -0,0 +1,153 @@
+
+describe('formatting', () => {
+    describe('number', () => {
+        it('returns value with undefined format', () => {
+            const formatter = getFormatter(undefined);
+            assert.isOk(formatter);
+
+            expect(formatter(0)).to.equal(0);
+            expect(formatter(1.766)).to.equal(1.766);
+            expect(formatter('foo')).to.equal('foo');
+            expect(isNaN(formatter(NaN))).to.equal(true);
+        });
+
+        describe('without nully handling / default locale', () => {
+            it('formats symbol', () => {
+                const formatter = getFormatter({
+                    locale: getLocale(),
+                    nully: getNully(),
+                    specifier: getSpecifier('$.2f')
+                });
+
+                assert.isOk(formatter);
+
+                expect(formatter(0)).to.equal('$0.00');
+                expect(formatter(1)).to.equal('$1.00');
+                expect(formatter(-1)).to.equal('-$1.00');
+                expect(formatter(1.23)).to.equal('$1.23');
+                expect(formatter(1.232)).to.equal('$1.23');
+                expect(formatter(1.239)).to.equal('$1.24');
+                expect(formatter(1766)).to.equal('$1766.00');
+                expect(formatter(''), 'Empty string case').to.equal('');
+                expect(formatter('foo'), 'Foo string case').to.equal('foo');
+                expect(formatter(true), 'Empty string case').to.equal(true);
+                expect(formatter(NaN), 'NaN case').to.equal('');
+                expect(formatter(Infinity), 'Infinity case').to.equal('');
+                expect(formatter(-Infinity), '-Infinity case').to.equal('');
+                expect(formatter(null as any), 'null case').to.equal('');
+                expect(formatter(undefined as any), 'undef case').to.equal('');
+            });
+        });
+
+        describe('with nully handling / default locale', () => {
+            it('formats significant digits and grouping separator', () => {
+                const formatter = getFormatter({
+                    locale: getLocale(),
+                    nully: 42.42,
+                    specifier: getSpecifier(',.2r')
+                });
+
+                assert.isOk(formatter);
+
+                expect(formatter(0)).to.equal('0.0');
+                expect(formatter(0.13)).to.equal('0.13');
+                expect(formatter(0.131)).to.equal('0.13');
+                expect(formatter(1.23)).to.equal('1.2');
+                expect(formatter(1.299)).to.equal('1.3');
+                expect(formatter(1299)).to.equal('1,300');
+                expect(formatter(1299431)).to.equal('1,300,000');
+                expect(formatter(''), 'Empty string case').to.equal('');
+                expect(formatter('foo'), 'Foo string case').to.equal('foo');
+                expect(formatter(true), 'Empty string case').to.equal(true);
+                expect(formatter(NaN), 'NaN case').to.equal('42');
```

Comment:
Ah interesting - if `nully` is a number it still has to pass through the formatter, that makes sense. Good call!